### PR TITLE
T257811: Add preview error handling

### PR DESCRIPTION
**Pull request by medied**
https://phabricator.wikimedia.org/T257811

Preview error handling following design's requirements:

    * Error handling state shows up when we are unable to show preview due to different errors(status code 204/500).
    * Error copy: There was an issue displaying this preview.
    * Users can close the error card either by tap close icon or area outside it.
    * Tap on CTA would take users to the article page.

Following logic from #31, I'm including `renderError` as part of methods available in preview.js

### CODE REVIEW

File: src/preview.js [hunk 1]
```diff
@@ -50,6 +50,29 @@ const renderPreview = ( lang, data, isTouch ) => {
...
+						<div class="wikipediapreview-header-closebtn"></div>
+					</div>
+					<div class="wikipediapreview-error">
+						<div class="wikipediapreview-error-body">
+							<div class="wikipediapreview-error-body-message">
+								<div class="wikipediapreview-error-body-icon"></div>
+								${msg( lang, 'preview-loading-error' )}
+							</div>
+							<div class="wikipediapreview-error-body-readon">
+								<a href=${`https://${lang}.m.wikipedia.org/wiki/${encodeURIComponent( title )}`}>${msg( lang, 'preview-loading-error-read' )}</a>
```
Comment (hueitan) [inline src/preview.js:70]: the target link should be _blank

File: style/preview.less [hunk 2]
```diff
@@ -215,6 +215,55 @@
...
+			height: 100%;
+			display: flex;
+			flex-direction: column;
+			justify-content: center;
+
+			&-message {
+				display: flex;
+				margin-left: 8%;
+				margin-right: 8%;
+				font-family: 'Lato', sans-serif;
```
Comment (hueitan) [inline style/preview.less:232]: we don't set font-family in the entire library, we use default font for all.

File: style/preview.less [hunk 3]
```diff
@@ -215,6 +215,55 @@
 		}
 	}
 
+	&-error {
```
Comment (hueitan) [inline style/preview.less:218]: remember to center the message (both horizontal and vertical), do check on the larget screen (ex. iPad size)

File: style/preview.less (hunk 2 above)
Comment (medied) [reply style/preview.less:232]: Do you mean I should remove the `font-family` lines?

**Comment by jpita** [comment]
@hueitan ur requested changes are done?
@medied can you give me a hint on how to test this?

**Comment by hueitan** [comment]
I'm still waiting for the text alignment confirmation here.
image

**Comment by medied** [comment]
I synced with Sudhanshu in a call today and we sorted out the text alignment, I have pushed this commit already. Huei/Pita you can now please proceed as usual

**Comment by jpita** [comment]
There is probably another race condition between showing the loading template and showing the error.
I can pass the mouse over the word (with a broken link) without letting it stay there and then the pop-up stays open.

Small GIF (400x364)

**Comment by medied** [comment]
Good catch @jpita! https://github.com/wikimedia/wikipedia-preview/pull/39/commits/636f0dcca115d0b2cc6be4734042748761f0ab88 should take care of that issue now, please feel free to test again

**Pull request merged**